apps/settings/js/panels/addon_details/addon_details.js

```diff
@@ -0,0 +1,169 @@
+      descriptionElement.textContent = description;
+    }
+    else {
+      // If there is no description, then display "no description" or similar.
+      descriptionElement.setAttribute('data-l10n-id', 'addon-no-description');
+    }
+
+
+    // Display the add-on developer name if there is one
+    var developerName = manifest.developer && manifest.developer.name;
+    var developerElement = $('#addon-developer');
+    if (developerName) {
+      developerElement.hidden = false;
+      l10n.setAttributes(developerElement, 'addon-developer', {
+        developerName: developerName
+      });
+    }
+    else {
+      developerElement.hidden = true;
+    }
+
+    // Get the list of targeted apps
+    $('#addon-targets').textContent = '';  // Clear any old content here
+    AddonManager.getAddonTargets(app).then(function(targets) {
+      var names = [];
+      targets.forEach(function(target) {
+        var manifest = new ManifestHelper(target.manifest ||
+                                          target.updateManifest);
+        names.push(manifest.name);
+      });
+
+      names.sort();
+
+      var container = $('#addon-targets');
+      names.forEach(function(name) {
+        var item = document.createElement('li');
+        var para = document.createElement('p');
+        para.textContent = name;
+        item.appendChild(para);
+        container.appendChild(item);
+      });
+
+      if (names.length === 0) {
+        $('#addon-targets').hidden = true;
+        $('#addon-no-targets').hidden = false;
+      }
+      else {
+        $('#addon-targets').hidden = false;
+        $('#addon-no-targets').hidden = true;
+      }
+    });
+
+    // Hook up the enable/disable toggle button
+    var toggle = $('#addon-enabled');
+    toggle.checked = AddonManager.isEnabled(app); // set initial state
+    toggle.onchange = function() {                // handle user changes
+      if (toggle.checked) {
+        AddonManager.enableAddon(app);
+      }
+      else {
+        AddonManager.disableAddon(app).then(function(status) {
+          // If an addon injected a script then we have to restart any
+          // affected apps. And if those apps are system apps, then we
+          // need to reboot the device.
+          if (status === 'reboot') {
+            // XXX We could do a confirm dialog here asking the user if
+            // they want to go ahead and reboot the device now.
+            Toaster.showToast({
+              messageL10nId: 'addon-reboot-required',
+              latency: 3000,
+              useTransition: true
+            });
+          }
+          else if (status === 'restart') {
+            Toaster.showToast({
+              messageL10nId: 'addon-restart-apps-to-disable',
+              latency: 3000,
+              useTransition: true
+            });
+          }
+        });
+      }
+    };
+
+    // The spec calls for a Share button in this panel, but we don't
```

Comment:
Maybe we can remove these for now if a sharing app will not be ready any time soon.